src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,67 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconSize } from "@components/Icon/Icon";
+import { ReactComponent as IconCaretDown } from "@components/Icon/Svg/CaretDown.svg";
+import { ReactComponent as IconCaretUp } from "@components/Icon/Svg/CaretUp.svg";
+import { Size } from "@utilities/enum";
+import { ReactElement, useState } from "react";
+import css from "./Dropdown.module.css";
+import DropdownMenu from "./DropdownMenu/DropdownMenu";
+import DropdownMenuItem, { MenuItem } from "./DropdownMenuItem/DropdownMenuItem";
+
+export interface DropdownProps {
+    menuItems: MenuItem[][];
+    onChange: (id?: string) => void;
+    activeItemId?: string;
+    placeholder?: string;
+    size?: Size.Small | Size.Large;
+}
+
+const getActiveItem = (menuItems: MenuItem[][], id: string) => menuItems.flat().find((item) => item.id === id) || null;
+
+export default function Dropdown({
+    menuItems,
+    onChange,
+    activeItemId = "",
+    placeholder = "",
+    size = Size.Small,
+}: DropdownProps): ReactElement<DropdownProps> {
+    const [open, setOpen] = useState(false);
+    const activeItem = getActiveItem(menuItems, activeItemId);
+
+    return (
+        <div className={css.wrapper}>
+            <button
+                data-test-id="dropdown-trigger"
+                onClick={() => setOpen(!open)}
+                className={`${css.trigger} ${!activeItem ? css.inactive : ""} ${size === Size.Large ? css.large : ""}`}
```

Comment:
There is also another way you could do that but not sure it is more readable
`[css.trigger, !activeItem && css.inactive, size === Size.Large && css.large].filter(Boolean).join(" " )`

Reply:
I have extracted it into a function as in the `MenuItem` component. It's more readable that way in my opinion.